src/Illuminate/Console/CacheCommandMutex.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Illuminate\Console;
+
+use Carbon\CarbonInterval;
+use Illuminate\Contracts\Cache\Factory as Cache;
+
+class CacheCommandMutex implements CommandMutex
+{
+    /**
+     * The cache store that should be used.
+     *
+     * @var string|null
+     */
+    public $store = null;
+
+    /**
+     * The cache factory implementation.
+     *
+     * @var \Illuminate\Contracts\Cache\Factory
+     */
+    public $cache;
+
+    /**
+     * Create a new command mutex.
+     *
+     * @param  \Illuminate\Contracts\Cache\Factory  $cache
+     */
+    public function __construct(Cache $cache)
+    {
+        $this->cache = $cache;
+    }
+
+    public function create($command)
+    {
+        return $this->cache->store($this->store)->add(
+            $this->commandMutexName($command),
+            true,
+            CarbonInterval::hour(),
+        );
+    }
+
+    public function exists($command)
+    {
+        return $this->cache->store($this->store)->has(
+            $this->commandMutexName($command)
+        );
+    }
+
+    public function release($command)
+    {
+        return $this->cache->store($this->store)->forget(
+            $this->commandMutexName($command)
+        );
+    }
+
+    /**
+     * @param  Command  $command
```

Comment:
We use FQN everywhere in DocBlocks.
```suggestion
     * @param  \Illuminate\Console\Command  $command
```